### PR TITLE
New profile view

### PR DESCRIPTION
**Pull request by muaathaw**
profile view is now done, any suggestion before i merge?

### CODE REVIEW

**Comment by EmEhPa** [comment]
Please merge main into your branch first.

File: src/views/ProfileView.vue [hunk 1]
```diff
@@ -1,9 +1,151 @@
...
+<script setup lang="ts">
+import { ref } from 'vue'
+
+const name = ref('')
+const username = ref('')
+const email = ref('')
+const password = ref('')
+const confirmPassword = ref('')
+
+// Method to handle profile updates
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:10]: this comment is not needed, the name of the function describes the purpose

File: src/views/ProfileView.vue [hunk 2]
```diff
@@ -1,9 +1,151 @@
...
+  }
+
+  // Handle profile update logic (send updated data to the server)
+  console.log('Profile updated with:', {
+    name: name.value,
+    username: username.value,
+    email: email.value,
+    password: password.value ? password.value : 'No password change'
+  })
+  // Add code to make an API call to update the user's profile
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:24]: remove GPT comment

File: src/views/ProfileView.vue [hunk 3]
```diff
@@ -1,9 +1,151 @@
...
+}
+
+// Method to handle account deletion
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:27]: same as before, its not needed

**Review by EmEhPa — CHANGES_REQUESTED** [review]

**Comment by EmEhPa** [comment]
See on Figma, maybe include so we the name will be visible

File: src/views/ProfileView.vue [hunk 4]
```diff
@@ -1,9 +1,151 @@
...
+const confirmPassword = ref('')
+
+// Method to handle profile updates
+const handleProfileUpdateClick = () => {
+  if (password.value && password.value !== confirmPassword.value) {
+    alert('Passwords do not match')
+    return
+  }
+
+  // Handle profile update logic (send updated data to the server)
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:17]: remove the console.log and comment and replace with a `TODO: <what to do in the future>`

File: src/views/ProfileView.vue [hunk 5]
```diff
@@ -1,9 +1,151 @@
...
+  // Add code to make an API call to update the user's profile
+}
+
+// Method to handle account deletion
+const handleAccountDeleteClick = () => {
+  const confirmation = confirm(
+    'Are you sure you want to delete your account? This action is irreversible.'
+  )
+  if (confirmation) {
+    // Add code to make an API call to delete the user's account
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:33]: again replace with `TODO` comment

File: src/views/ProfileView.vue [hunk 6]
```diff
@@ -1,9 +1,151 @@
...
+    console.log('Account deleted')
+  }
+}
+
+// Method to handle declining (canceling) changes
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:38]: not needed

File: src/views/ProfileView.vue [hunk 7]
```diff
@@ -1,9 +1,151 @@
...
+const handleDeclineChangesClick = () => {
+  const confirmation = confirm(
+    'Are you sure you want to cancel the changes? Unsaved changes will be lost.'
+  )
+  if (confirmation) {
+    // Reset the form fields to their previous values or leave empty if no previous values
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:44]: not needed aswell, code explains itself

File: src/views/ProfileView.vue [hunk 8]
```diff
@@ -1,9 +1,151 @@
...
   <section class="section">
-    <div class="container">
-      <h1 class="title has-text-centered">Profile</h1>
+    <div class="Thecontainer">
+      <div class="box">
+        <h2>
+          <slot name="heading">Edit Profile</slot>
+        </h2>
+
+        <!-- Profile form content -->
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:63]: remove

File: src/views/ProfileView.vue [hunk 9]
```diff
@@ -1,9 +1,151 @@
...
+        <form @submit.prevent="handleProfileUpdateClick">
+          <!-- Name Field -->
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:65]: remove

File: src/views/ProfileView.vue [hunk 10]
```diff
@@ -1,9 +1,151 @@
...
+          <div class="field">
+            <label class="label">Name</label>
+            <div class="control">
+              <input type="text" v-model="name" class="input" placeholder="Name" required />
+            </div>
+          </div>
+
+          <!-- Username Field -->
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:73]: remove

File: src/views/ProfileView.vue [hunk 11]
```diff
@@ -1,9 +1,151 @@
...
+          <div class="field">
+            <label class="label">Username</label>
+            <div class="control">
+              <input type="text" v-model="username" class="input" placeholder="Username" required />
+            </div>
+          </div>
+
+          <!-- Email Field -->
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:81]: remove

File: src/views/ProfileView.vue [hunk 12]
```diff
@@ -1,9 +1,151 @@
...
+          <div class="field">
+            <label class="label">Email</label>
+            <div class="control">
+              <input type="email" v-model="email" class="input" placeholder="Email" required />
+            </div>
+          </div>
+
+          <!-- Password Fields (optional) -->
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:89]: remove

File: src/views/ProfileView.vue [hunk 13]
```diff
@@ -1,9 +1,151 @@
...
+                type="password"
+                v-model="confirmPassword"
+                class="input"
+                placeholder="Confirm New Password"
+                minlength="8"
+              />
+            </div>
+          </div>
+
+          <!-- Submit Button -->
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:116]: remove

File: src/views/ProfileView.vue [hunk 14]
```diff
@@ -1,9 +1,151 @@
...
+          <div class="columns is-centered mt-5">
+            <div class="column is-narrow">
+              <button type="submit" class="button is-primary custom-button">Update Profile</button>
+            </div>
+          </div>
+        </form>
+
+        <!-- Additional Actions: Decline and Delete Account -->
```
Comment (retrokiller543) [inline src/views/ProfileView.vue:124]: remove

**Review by retrokiller543 — CHANGES_REQUESTED** [review]

File: src/views/ProfileView.vue (hunk 1 above)
Comment (retrokiller543) [inline src/views/ProfileView.vue:10]: Not needed, function name describes iteself

File: src/views/ProfileView.vue [hunk 15]
```diff
@@ -24,11 +23,38 @@ const handleProfileUpdateClick = () => {
...
+    'Are you sure you want to delete your account? This action is irreversible.'
+  )
+  if (confirmation) {
+    // Add code to make an API call to delete the user's account
+    console.log('Account deleted')
+  }
+}
+
+// Method to handle declining (canceling) changes
+const handleDeclineChangesClick = () => {
```
[the remark this answers is not in the document]
Comment (muaathaw) [reply src/views/ProfileView.vue:39]: NO! it´s just a function to restore the changes, in case the user changes something wrong or regrets it

**Review by retrokiller543 — APPROVED** [review]
Looks good, there is a comment that might have been missed but im happy with the changes made

**Review by EmEhPa — APPROVED** [review]
Ok! The main design is there so we can merge. We will during next team team implement the logic of this view

**Pull request merged**